### DISCUSSION TITLE
How to override pimcore portal engine services?

### DISCUSSION THREAD

**Discussion by nishitsureka** [pimcore/pimcore#15961] [Q&A / Support]
Hi,
I tried to override portal engine services but unable to do so.
I have created Dataobject data pool page to list all the product data.
But somehow it is showing only published product in data pool page.
So i tried to override the services to disable published code.
but it is not overriding.

so, simply is it possible to override services of portal engine?

**Comment by dvesh3** [comment]
yes, you can override services of portal engine just like any other Symfony service. Please share your implementation so that we can help you with the issue.

**Comment by dpfaffenbauer** [comment]
that might help: https://symfony.com/doc/current/service_container/service_decoration.html
